src/index.ts

```diff
@@ -304,52 +312,26 @@ export function HTPipePreAuthorize<
           : TContextInRight & TContextInLeft
       ) => {
         const leftOut = left.preAuthorize(context);
-        const leftOutIsBoolean = leftOut === true || leftOut === false;
-        if (leftOutIsBoolean) {
-          if (leftOut) {
-            const rightOut = right.preAuthorize(context);
-            const rightOutIsBoolean = rightOut === true || rightOut === false;
-            if (rightOutIsBoolean) {
-              if (rightOut) {
-                return rightOut;
-              } else {
-                return false;
-              }
-            } else {
-              if (Object.keys(rightOut).length > 0) {
-                return rightOut;
-              } else {
-                return false;
-              }
-            }
-          } else {
-            return false;
-          }
-        } else {
-          if (Object.keys(leftOut).length > 0) {
-            const rightIn = {
-              ...context,
-              ...leftOut,
-            };
-            const rightOut = right.preAuthorize(rightIn);
-            const rightOutIsBoolean = rightOut === true || rightOut === false;
-            if (rightOutIsBoolean) {
-              if (rightOut) {
-                return rightOut;
-              } else {
-                return false;
-              }
-            } else {
-              if (Object.keys(rightOut).length > 0) {
-                return rightOut;
-              } else {
-                return false;
-              }
-            }
-          } else {
-            return false;
-          }
+        const leftPassed =
+          leftOut === true ||
+          (leftOut &&
+            typeof leftOut === 'object' &&
+            Object.keys(leftOut).length > 0);
+        if (!leftPassed) {
+          return false;
         }
+        const leftContextOut = leftPassed === true ? {} : leftOut;
+        const rightIn = {
+          ...context,
+          ...leftContextOut,
+        };
+        const rightOut = right.preAuthorize(rightIn);
```

Comment:
returning rightPassed will lose context if right wants to return some ... why not skip rightPassed and just return rightOut?
- [ ] fix lost context

Reply:
done